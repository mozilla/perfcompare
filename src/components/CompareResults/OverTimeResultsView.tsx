import { useEffect } from 'react';

import Grid from '@mui/material/Grid';
import { useLoaderData } from 'react-router-dom';
import { style } from 'typestyle';

import { useAppSelector } from '../../hooks/app';
import { SearchContainerStyles, background } from '../../styles';
import CompareOverTime from '../Search/CompareOverTime';
import { LinkToHome } from '../Shared/LinkToHome';
import PerfCompareHeader from '../Shared/PerfCompareHeader';
import type { LoaderReturnValue } from './overTimeLoader';
import ResultsMain from './ResultsMain';

interface ResultsViewProps {
  title: string;
}
function ResultsView(props: ResultsViewProps) {
  const {
    newRevsInfo,
    frameworkId,
    intervalValue,

    baseRepo,
    newRepos,
  } = useLoaderData() as LoaderReturnValue;
  const newRepo = newRepos[0];
  const { title } = props;
  const themeMode = useAppSelector((state) => state.theme.mode);
  const styles = {
    container: style({
      backgroundColor: background(themeMode),
    }),
  };

  const sectionStyles = SearchContainerStyles(themeMode, /* isHome */ false);

  useEffect(() => {
    document.title = title;
  }, [title]);

  return (
    <div
      className={styles.container}
      data-testid='beta-version-compare-overtime-results'
    >
      <PerfCompareHeader />
      <section className={sectionStyles.container}>
        <LinkToHome />
        <CompareOverTime
          hasEditButton={true}
          newRevs={newRevsInfo ?? []}
          isBaseSearch={true}
          expandBaseComponent={() => null}
          frameworkIdVal={frameworkId}
          intervalValue={intervalValue}
          baseRepo={baseRepo}
          newRepo={newRepo}
        />
      </section>
      <Grid container alignItems='center' justifyContent='center'>
        <Grid item xs={12}>
          <ResultsMain />
        </Grid>
      </Grid>
    </div>
  );
}

export default ResultsView;
