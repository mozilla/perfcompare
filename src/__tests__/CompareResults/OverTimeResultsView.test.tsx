/* eslint-disable jest/no-disabled-tests */
import type { ReactElement } from 'react';

import { loader } from '../../components/CompareResults/overTimeLoader';
import OverTimeResultsView from '../../components/CompareResults/OverTimeResultsView';
import { Strings } from '../../resources/Strings';
import getTestData from '../utils/fixtures';
import {
  renderWithRouter,
  screen,
  FetchMockSandbox,
} from '../utils/test-utils';

function renderWithRoute(component: ReactElement) {
  const { testCompareData, testData } = getTestData();
  (window.fetch as FetchMockSandbox)
    .get(
      'begin:https://treeherder.mozilla.org/api/perfcompare/results/',
      testCompareData,
    )
    .get('begin:https://treeherder.mozilla.org/api/project/', {
      results: [testData[0]],
    });

  return renderWithRouter(component, {
    route: '/compare-over-time-results/',
    search:
      '?baseRepo=try&selectedTimeRange=86400&newRev=try&newRepo=try&framework=1',
    loader,
  });
}

jest.mock('../../utils/location');
// const mockedGetLocationOrigin = getLocationOrigin as jest.Mock;

describe('Results View', () => {
  it('The table should match snapshot and other elements should be present in the page', async () => {
    renderWithRoute(
      <OverTimeResultsView title={Strings.metaData.pageTitle.results} />,
    );

    expect(await screen.findByRole('table')).toMatchSnapshot();
    const link = screen.getByRole('link', { name: /link to home/i });
    expect(link).toBeInTheDocument();
  });
});
