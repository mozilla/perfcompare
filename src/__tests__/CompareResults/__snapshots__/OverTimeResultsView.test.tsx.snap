// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results View The table should match snapshot and other elements should be present in the page 1`] = `
<div
  class="MuiBox-root css-hshm0p"
  data-testid="results-table"
  role="table"
>
  <div
    class="MuiBox-root css-1rw0zd7"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 f1wpas00 css-1e9xsub-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-fo77co-MuiGrid-root"
      >
        <form
          aria-label="Search by title, platform, revision or options"
          class="MuiBox-root css-1wxqtyp"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1cocm2p-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart MuiInputBase-adornedEnd css-m7khg8-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                aria-label="Search by title, platform, revision or options"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-rg83rj-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r3:"
                placeholder="Search results"
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="Clear the search input"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-151ffkv-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                    data-testid="CloseIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </form>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1t7ybvx-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall framework-dropdown-select css-1iil27j-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            data-testid="framework-select"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="Framework"
              aria-labelledby="mui-component-select-framework"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jaypmb-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-component-select-framework"
              role="button"
              tabindex="0"
            >
              talos
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              name="framework"
              tabindex="-1"
              value="1"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-w4piji-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1t7ybvx-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-0"
          data-testid="revision-select"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall f96or0t css-1iil27j-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jaypmb-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              <div
                class="fbc330 MuiBox-root css-0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                    data-testid="SortIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
                    />
                  </svg>
                </svg>
                All revisions
              </div>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="all-revisions"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-w4piji-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1t7ybvx-MuiGrid-root"
      >
        <div
          class="f6hg2jo"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-iendvr-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Download JSON
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="f1wqnr25 fr2n4tx"
      data-testid="table-header"
      role="row"
    >
      <div
        class="cell platform-header"
        role="columnheader"
      >
        <button
          aria-haspopup="true"
          aria-label="Platform (Click to filter values)"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-gsklu7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Platform
          <svg
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-reqpbo-MuiSvgIcon-root"
            data-testid="FilterListIcon"
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
            />
            <title>
              No active filters
            </title>
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="cell base-header"
        role="columnheader"
      >
        Base
      </div>
      <div
        class="cell comparisonSign-header"
        role="columnheader"
      />
      <div
        class="cell new-header"
        role="columnheader"
      >
        New
      </div>
      <div
        class="cell status-header"
        role="columnheader"
      >
        <button
          aria-haspopup="true"
          aria-label="Status (Click to filter values)"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-gsklu7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Status
          <svg
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-reqpbo-MuiSvgIcon-root"
            data-testid="FilterListIcon"
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
            />
            <title>
              No active filters
            </title>
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="cell delta-header"
        role="columnheader"
      >
        Delta(%)
      </div>
      <div
        class="cell confidence-header"
        role="columnheader"
      >
        <button
          aria-haspopup="true"
          aria-label="Confidence (Click to filter values. Some filters are active.)"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-gsklu7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Confidence
          <svg
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1yguepm-MuiSvgIcon-root"
            data-testid="FilterListIcon"
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
            />
            <title>
              Some filters are active
            </title>
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="cell runs-header"
        role="columnheader"
      >
        Total Runs
      </div>
      <div
        class="cell buttons-header"
        role="columnheader"
      />
      <div
        class="cell expand-header"
        role="columnheader"
      />
    </div>
  </div>
  <div
    data-virtuoso-scroller="true"
    style="position: relative; height: 100px;"
  >
    <div
      data-viewport-type="window"
      style="width: 100%; height: 100%; position: absolute; top: 0px;"
    >
      <div
        data-testid="virtuoso-item-list"
        style="box-sizing: border-box; margin-top: 0px; padding-top: 0px; padding-bottom: 0px;"
      >
        <div
          data-index="0"
          data-item-index="0"
          data-known-size="100"
          style="overflow-anchor: none;"
        >
          <div
            class="f1otqxop"
            role="rowgroup"
          >
            <div
              class="fx9q187"
            >
              <div
                class="f897l9y"
              >
                <strong>
                  <a
                    aria-label="link to suite documentation"
                    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-g8z9fn-MuiTypography-root-MuiLink-root"
                    href="https://firefox-source-docs.mozilla.org/testing/perfdocs/talos.html#a11yr"
                    target="_blank"
                  >
                    a11yr
                  </a>
                   dhtml.html
                </strong>
                 
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qhous5-MuiTypography-root-MuiLink-root"
                  href="https://treeherder.mozilla.org/jobs?repo=mozilla-central&revision=spam"
                  target="_blank"
                  title="open treeherder view for spam"
                >
                  spam
                </a>
              </div>
              <div
                class="f1oxa67k"
              >
                <span
                  class="f1rpr1m5"
                >
                  opt
                </span>
                <span
                  class="f1rpr1m5"
                >
                  e10s
                </span>
                <span
                  class="f1rpr1m5"
                >
                  fission
                </span>
                <span
                  class="f1rpr1m5"
                >
                  stylo
                </span>
                <span
                  class="f1rpr1m5"
                >
                  webrender
                </span>
              </div>
            </div>
            <div>
              <div
                class="revisionRow f1b00bec fp4ugv1 MuiBox-root css-1am90bh"
                role="row"
              >
                <div
                  class="platform cell"
                  role="cell"
                >
                  <div
                    aria-label="windows10-64-shippable-qr"
                    class="platform-container"
                    data-mui-internal-clone-element="true"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 30 30"
                    >
                      <path
                        d="M13.875 7.719l-2.281 7.906s-2.594-1.813-4.5-1.625c-1.875 0.188-4.406 1.125-4.406 1.125l2.313-7.969s4.781-2.563 8.875 0.563zM12.813 16.375l2.281-7.844c0.688 0.469 1.906 1.313 3.875 1.563 1.969 0.219 5.031-1.031 5.031-1.031l-2.281 7.906c-0.344 0.125-1.781 0.938-4.406 1.094-2.656 0.156-4.5-1.688-4.5-1.688zM0 24.469l2.281-7.969s2.563-1.188 4.719-1.031c2.125 0.125 3.844 1.313 4.188 1.594l-2.281 7.969s-2.625-1.719-4.438-1.563-3.406 0.469-4.469 1zM10.125 25.719l2.281-7.906s2.438 1.594 4.031 1.531c1.594-0.031 2.688-0.125 4.844-0.969h0.031c-0.156 0.406-2.281 7.844-2.281 7.844s-2.5 1.188-4.813 1.063c-2.281-0.094-3.656-1.406-4.094-1.563z"
                      />
                       
                    </svg>
                    <span>
                      Windows
                    </span>
                  </div>
                </div>
                <div
                  class="base-value cell"
                  role="cell"
                >
                   
                  643.54
                   
                  ms
                   
                </div>
                <div
                  class="comparison-sign cell"
                  role="cell"
                >
                  &gt;
                </div>
                <div
                  class="new-value cell"
                  role="cell"
                >
                   
                  628.09
                   
                  ms
                </div>
                <div
                  class="status cell"
                  role="cell"
                >
                  <span
                    class="status-hint "
                  >
                    -
                  </span>
                </div>
                <div
                  class="delta cell"
                  role="cell"
                >
                   
                  -2.4
                   %
                   
                </div>
                <div
                  class="confidence cell"
                  role="cell"
                >
                   
                  High
                   
                </div>
                <div
                  class="total-runs cell"
                  role="cell"
                >
                  <span>
                    <span
                      title="Base runs"
                    >
                      B:
                    </span>
                    <strong>
                      1
                    </strong>
                  </span>
                  <span>
                    <span
                      title="New runs"
                    >
                      N:
                    </span>
                    <strong>
                      1
                    </strong>
                  </span>
                </div>
                <div
                  class="row-buttons cell"
                >
                  <div
                    class="graph"
                    role="cell"
                  >
                    <div
                      class="graph-link-button-container"
                    >
                      <a
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-1itd531-MuiButtonBase-root-MuiIconButton-root"
                        href="https://treeherder.mozilla.org/perfherder/graphs?highlightedRevisions=1e32960788ea&highlightedRevisions=6a1f133815a1&series=mozilla-central%2C3a6d4b6178678c3113c980b378bb83b2629926a0%2C1%2C1&timerange=5184000"
                        tabindex="0"
                        target="_blank"
                        title="open the evolution graph for this job in treeherder"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                          data-testid="TimelineIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M23 8c0 1.1-.9 2-2 2-.18 0-.35-.02-.51-.07l-3.56 3.55c.05.16.07.34.07.52 0 1.1-.9 2-2 2s-2-.9-2-2c0-.18.02-.36.07-.52l-2.55-2.55c-.16.05-.34.07-.52.07s-.36-.02-.52-.07l-4.55 4.56c.05.16.07.33.07.51 0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2c.18 0 .35.02.51.07l4.56-4.55C8.02 9.36 8 9.18 8 9c0-1.1.9-2 2-2s2 .9 2 2c0 .18-.02.36-.07.52l2.55 2.55c.16-.05.34-.07.52-.07s.36.02.52.07l3.55-3.56C19.02 8.35 19 8.18 19 8c0-1.1.9-2 2-2s2 .9 2 2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                    </div>
                  </div>
                  <div
                    class="retrigger-button"
                    data-testid="retrigger-jobs-button"
                    role="cell"
                  >
                    <div
                      class="retrigger-button-container"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-1itd531-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        title="retrigger jobs"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                          data-testid="RefreshOutlinedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="expand-button cell"
                  role="cell"
                >
                  <div
                    class="expand-button-container"
                    data-testid="expand-revision-button"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-1itd531-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      title="expand this row"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                        data-testid="ExpandMoreIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="revisionRow f1b00bec fp4ugv1 MuiBox-root css-1am90bh"
                role="row"
              >
                <div
                  class="platform cell"
                  role="cell"
                >
                  <div
                    aria-label="windows10-64-qr"
                    class="platform-container"
                    data-mui-internal-clone-element="true"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 30 30"
                    >
                      <path
                        d="M13.875 7.719l-2.281 7.906s-2.594-1.813-4.5-1.625c-1.875 0.188-4.406 1.125-4.406 1.125l2.313-7.969s4.781-2.563 8.875 0.563zM12.813 16.375l2.281-7.844c0.688 0.469 1.906 1.313 3.875 1.563 1.969 0.219 5.031-1.031 5.031-1.031l-2.281 7.906c-0.344 0.125-1.781 0.938-4.406 1.094-2.656 0.156-4.5-1.688-4.5-1.688zM0 24.469l2.281-7.969s2.563-1.188 4.719-1.031c2.125 0.125 3.844 1.313 4.188 1.594l-2.281 7.969s-2.625-1.719-4.438-1.563-3.406 0.469-4.469 1zM10.125 25.719l2.281-7.906s2.438 1.594 4.031 1.531c1.594-0.031 2.688-0.125 4.844-0.969h0.031c-0.156 0.406-2.281 7.844-2.281 7.844s-2.5 1.188-4.813 1.063c-2.281-0.094-3.656-1.406-4.094-1.563z"
                      />
                       
                    </svg>
                    <span>
                      Windows
                    </span>
                  </div>
                </div>
                <div
                  class="base-value cell"
                  role="cell"
                >
                   
                  643.54
                   
                  ms
                   
                </div>
                <div
                  class="comparison-sign cell"
                  role="cell"
                >
                  &gt;
                </div>
                <div
                  class="new-value cell"
                  role="cell"
                >
                   
                  628.09
                   
                  ms
                </div>
                <div
                  class="status cell"
                  role="cell"
                >
                  <span
                    class="status-hint "
                  >
                    -
                  </span>
                </div>
                <div
                  class="delta cell"
                  role="cell"
                >
                   
                  -2.4
                   %
                   
                </div>
                <div
                  class="confidence cell"
                  role="cell"
                >
                   
                   
                </div>
                <div
                  class="total-runs cell"
                  role="cell"
                >
                  <span>
                    <span
                      title="Base runs"
                    >
                      B:
                    </span>
                    <strong>
                      1
                    </strong>
                  </span>
                  <span>
                    <span
                      title="New runs"
                    >
                      N:
                    </span>
                    <strong>
                      1
                    </strong>
                  </span>
                </div>
                <div
                  class="row-buttons cell"
                >
                  <div
                    class="graph"
                    role="cell"
                  >
                    <div
                      class="graph-link-button-container"
                    >
                      <a
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-1itd531-MuiButtonBase-root-MuiIconButton-root"
                        href="https://treeherder.mozilla.org/perfherder/graphs?highlightedRevisions=1e32960788ea&highlightedRevisions=6a1f133815a1&series=mozilla-central%2C3a6d4b6178678c3113c980b378bb83b2629926a0%2C1%2C1&timerange=5184000"
                        tabindex="0"
                        target="_blank"
                        title="open the evolution graph for this job in treeherder"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                          data-testid="TimelineIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M23 8c0 1.1-.9 2-2 2-.18 0-.35-.02-.51-.07l-3.56 3.55c.05.16.07.34.07.52 0 1.1-.9 2-2 2s-2-.9-2-2c0-.18.02-.36.07-.52l-2.55-2.55c-.16.05-.34.07-.52.07s-.36-.02-.52-.07l-4.55 4.56c.05.16.07.33.07.51 0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2c.18 0 .35.02.51.07l4.56-4.55C8.02 9.36 8 9.18 8 9c0-1.1.9-2 2-2s2 .9 2 2c0 .18-.02.36-.07.52l2.55 2.55c.16-.05.34-.07.52-.07s.36.02.52.07l3.55-3.56C19.02 8.35 19 8.18 19 8c0-1.1.9-2 2-2s2 .9 2 2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                    </div>
                  </div>
                  <div
                    class="retrigger-button"
                    data-testid="retrigger-jobs-button"
                    role="cell"
                  >
                    <div
                      class="retrigger-button-container"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-1itd531-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        title="retrigger jobs"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                          data-testid="RefreshOutlinedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="expand-button cell"
                  role="cell"
                >
                  <div
                    class="expand-button-container"
                    data-testid="expand-revision-button"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-1itd531-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      title="expand this row"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                        data-testid="ExpandMoreIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
